File: app/src/main/java/prudential/pobmobilecustomerappandroid/utils/AsyncTaskCustom.java
package prudential.pobmobilecustomerappandroid.utils;

import android.os.AsyncTask;

public class AsyncTaskCustom extends AsyncTask<Void, Void, Object> {

	
	//############################################//
	//#GET SET
	//############################################//

	
	private Call call;
	private Object resp;
	
	public Call getCall() {
		return call;
	}

	public void setCall(Call call) {
		this.call = call;
	}
	
	
	//############################################//
	//#EXECUTE
	//############################################//


	@Override
	protected Object doInBackground(Void... params) {

		return call.onRun();
	}

	public Object getResp() {
		return resp;
	}

	public void setResp(Object resp) {
		this.resp = resp;
	}

	

	@Override
	protected void onPostExecute(Object result) {
		call.onComplete(result);
	}

	// ###################################################//
	// #INTERFACE
	// ###################################################//

	public interface Call {

		public Object onRun();
		public void onComplete(Object result);

	}

	

}

File: app/src/main/java/prudential/pobmobilecustomerappandroid/utils/model/AsyncTaskCustom.kt
package prudential.pobmobilecustomerappandroid.utils.model

import android.os.AsyncTask

class AsyncTaskCustom() : AsyncTask<Void?, Void?, Any>() {

    var call: Call? = null
    var resp: Any? = null

    override fun onPostExecute(result: Any) {
        call!!.onComplete(result)
    }

    interface Call {
        fun onRun(): Any
        fun onComplete(result: Any?)
    }

    override fun doInBackground(vararg params: Void?): Any? {
        return call?.onRun()
    }
}